odoo_backup_sh/models/odoo_backup_sh.py

```diff
@@ -16,67 +21,289 @@
-                                 aws_secret_access_key=credentials['amazon_secret_access_key'])
-        ts = datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S")
-        filename = "%s|%s" % (filename, ts)
-        s3_path_key = '%s/%s' % (credentials['oauth_uid'], filename)
+        cloud_params = BackupController().get_local_cloud_params()
+        s3_client = boto3.client('s3', aws_access_key_id=cloud_params['amazon_access_key'],
+                                 aws_secret_access_key=cloud_params['amazon_secret_access_key'])
+        dt = datetime.utcnow()
+        ts = dt.strftime("%Y-%m-%d_%H-%M-%S")
+        s3_backup_path = '%s/%s.%s%s' % (cloud_params['oauth_uid'], name, ts, backup_name_suffix)
+        info_file = tempfile.TemporaryFile()
+        info_file.write('[common]\n'.encode())
+        info_file_content = {
+            'database': name,
+            'encrypted': True if '.enc' in backup_name_suffix else False,
+            'upload_datetime': ts,
+        }
+        for key, value in info_file_content.items():
+            line = key + ' = ' + str(value) + '\n'
+            info_file.write(line.encode())
+        info_file.seek(0)
+        s3_info_file_path = '%s/%s.%s.info' % (cloud_params['oauth_uid'], name, ts)
+        # Create new record with backup info data
+        info_file_content['upload_datetime'] = dt
+        self.env['odoo_backup_sh.backup_info'].create(info_file_content)
+        # Upload two backup objects to AWS S3
         try:
-            s3_client.put_object(Body=dump_stream, Bucket=credentials['amazon_bucket_name'], Key=s3_path_key)
+            s3_client.put_object(Body=dump_stream, Bucket=cloud_params['amazon_bucket_name'], Key=s3_backup_path)
+            s3_client.put_object(Body=info_file, Bucket=cloud_params['amazon_bucket_name'], Key=s3_info_file_path)
         except botocore.exceptions.ClientError as e:
             raise exceptions.ValidationError(_("Amazon error: ") + e.response['Error']['Message'])
+        self.update_info(cloud_params)
         return None
 
 
-class Backup(models.Model):
-    _name = 'odoo_backup_sh.backup'
+class BackupConfigCron(models.Model):
+    _name = 'odoo_backup_sh.config.cron'
+    _inherits = {'ir.cron': 'ir_cron_id'}
+    _description = 'Backup Configuration Scheduled Actions'
+
+    backup_config_id = fields.Many2one(
+        'odoo_backup_sh.config', string='Backup Configuration', required=True)
+    ir_cron_id = fields.Many2one('ir.cron', string='Scheduled Action', required=True, ondelete='restrict')
+
+    @api.model
+    def create(self, vals):
+        database = self.env['odoo_backup_sh.config'].browse(vals['backup_config_id']).database
+        vals.update({
+            'name': 'Odoo-backup.sh: Backup the "%s" database and send it to the remote storage' % database,
+            'model_id': self.env.ref('odoo_backup_sh.model_odoo_backup_sh_config').id,
+            'state': 'code',
+            'numbercall': -1,
+            'doall': False,
```

Comment:
This could be done via ``fields(default=...)``

Reply:
We discussed it verbally and decided to leave as is.